### ISSUE TITLE
Memory Leak From Kafka Producer with kafka-net 0.9.0.49

### ISSUE THREAD

**Issue by mt830813** [Jroland/kafka-net#69]
using memory profiling tool to make and compare snapshots,
second snapshot is 20 minutes later than the first one.
full gc collect before create snapshot.

below print is the remained objects from first to second.

In my code,I only using client.SendMessageAsync to produce message , but with out wait method.

![image](https://cloud.githubusercontent.com/assets/5535178/9457708/bc7c3a92-4b18-11e5-9548-48c165f47dd4.png)

is there any way to release those bytes array? or i using it wrong way?
looking forward  someone to help me.

**Comment by mt830813** [comment]
After Reading Source code,found this object maintain a 500 length queue name <CompletedNetworkWriteStatistics> to hold already send data.
When message too big like my case,(around 10M per message) may cause memory leak.
Still try to figure out why maintain this queue.

**Comment by EranOfer** [comment]
you can try disable Statistics Tracker and see if there any improvement.
in class KafkaTcpSocket try to comment
StatisticsTracker.QueueNetworkWrite(_endpoint, payload);
StatisticsTracker.CompleteNetworkWrite(sendTask.Payload, sw.ElapsedMilliseconds, failed);

try to look on NetworkWriteStatistic it contain the Payload and it may cause some of the problems.
please keep me updated if it help you.

**Comment by mt830813** [comment]
Yes,I Changed the source Code, like below and fixed the problem.
```c#
private static readonly ConcurrentCircularBuffer<NetworkWriteStatistic> CompletedNetworkWriteStatistics = new ConcurrentCircularBuffer<NetworkWriteStatistic>(1);
```
Thanks for your reply. Although it's a little bit too late :)
Any way,Is there any config to disable this tracker? or maybe using some variable to store key parameters instead of maintain full data in your object .

**Comment by EranOfer** [comment]
hi,
i am working on a new fork you can check it out .(it still work in progress )
you can check it out and open the issue there if they exists in my version ,i will answer more quickly and fix them .

because i work on another version some issue don't exists in my code so i rarely answer issues here.

https://github.com/EranOfer/KafkaNetClient
